espnet/asr/pytorch_backend/asr.py

```diff
@@ -447,6 +447,12 @@ def train(args):
         else:
             model, optimizer = amp.initialize(model, optimizer, opt_level=args.train_dtype)
         use_apex = True
+
+        if args.ctc_type == "builtin":
+            from espnet.nets.pytorch_backend.ctc import CTC
+            amp.register_float_function(CTC, "loss_fn")
+            amp.init()
+            logging.warning('register pytorch builtin ctc as float function')
```

Comment:
```suggestion
        from espnet.nets.pytorch_backend.ctc import CTC
        amp.register_float_function(CTC, "loss_fn")
        amp.init()
        logging.warning('register ctc as float function')
```
Maybe we do not need to limit this to "builtin" because warpctc does not support float16 too.

Reply:
then the loss_fn of CTC could be simplified, isn't it?

Reply:
Yes if the apex.amp is always enabled. However, I would like to keep them because of  the disabled case and the latest pytorch amp.